pkg/service/service.go

```diff
@@ -1454,6 +1454,9 @@ func (s *Service) updateBackendsCacheLocked(svc *svcInfo, backends []*lb.Backend
 
 	for i, backend := range backends {
 		hash := backend.L3n4Addr.Hash()
+		if _, found := backendSet[hash]; found {
+			return nil, nil, nil, fmt.Errorf("duplicated backends are not allowed")
+		}
```

Comment:
I'm not sure if this is what we want here. the `hash` on its own prevents on having duplicated backends. That's why we are storing it by hash.

Reply:
As you know, `hash` does not care protocols.
If you have better idea, let me know and  I will appreciate.